### PR TITLE
Change cli init to new

### PR DESCRIPTION
**Pull request by ngfgrant**
A small rename of `init` cli command group to `new`
-----------------

* [x] Wrote [good commit messages][1].
* [x] [Squashed related commits together][2] after the changes have been approved.
* [x] Commit message starts with `[Resolve #issue-number]` (if a related issue exists).
* [x] Added unit tests.
* [x] Added integration tests (if applicable).
* [x] All unit tests (`make test`) are passing.
* [x] Used the same coding conventions as the rest of the project.
* [x] The new code doesn't generate flake8 (`make lint`) offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -424,7 +424,7 @@ def test_status_with_stack(self):
         assert result.exit_code == 0
         assert result.output == "mock-stack: status\n"
 
-    def test_init_project_non_existant(self):
+    def test_new_project_non_existant(self):
```
Comment (ngfgrant) [inline tests/test_cli.py]: spelling error - existent

File: tests/test_cli.py [hunk 2]
```diff
@@ -446,7 +446,7 @@ def test_init_project_non_existant(self):
 
             assert config == defaults
 
-    def test_init_project_already_exist(self):
+    def test_new_project_already_exist(self):
```
Comment (ngfgrant) [inline tests/test_cli.py]: rn -> test_new_project_already_exists

**Comment by connec** [comment]
This needs rebased to remove most of the commits (except the last one, with the actual rename). It should probably also target `v2-rc`.

**Comment by ngfgrant** [comment]
Rebased for the release candidate.

**Pull request merged**